### PR TITLE
Fix sampling and scoring device bugs

### PR DESCRIPTION
**Pull request by naailkhan28**
As mentioned in issue #487 there is still some a CUDA device bug in `GVPTransformer.model.sample` - this PR fixes that plus a bug in `esm.inverse_folding.util.get_sequence_loss`

### CODE REVIEW

File: esm/inverse_folding/util.py [hunk 1]
```diff
@@ -112,10 +112,10 @@ def get_sequence_loss(model, alphabet, coords, seq):
     coords, confidence, strs, tokens, padding_mask = batch_converter(
         batch, device=device)
 
-    prev_output_tokens = tokens[:, :-1]
+    prev_output_tokens = tokens[:, :-1].to(device)
```
Comment (brianhie) [inline esm/inverse_folding/util.py:115]: @naailkhan28 what is the rationale for this change? Passing `device` into `model.forward()` seems like a more general fix than moving the tensor to the device beforehand.

File: esm/inverse_folding/gvp_transformer.py [hunk 2]
```diff
@@ -118,6 +118,10 @@ def sample(self, coords, partial_seq=None, temperature=1.0, confidence=None, dev
         # Run encoder only once
         encoder_out = self.encoder(batch_coords, padding_mask, confidence)
         
+        #Make sure all tensors are on the same device if a GPU is present
```
Comment (brianhie) [inline esm/inverse_folding/gvp_transformer.py:121]: Nit: Add a space between the `#` and the comment to match style

File: esm/inverse_folding/util.py (hunk 1 above)
Comment (brianhie) [reply esm/inverse_folding/util.py:115]: Actually, removing `device` from `model.forward()` is needed. My evaluation setup was improperly configured, so I was not catching the error.

File: esm/inverse_folding/gvp_transformer.py (hunk 2 above)
Comment (naailkhan28) [reply esm/inverse_folding/gvp_transformer.py]: @brianhie  Done! 

Scoring and sampling should work now - my setup wasn't catching the error too which I didn't realize until I investigated the linked issue yesterday :)

**Review by brianhie — APPROVED** [review]
LGTM
